text/0030-asc-desc-criterion.md

```diff
@@ -0,0 +1,95 @@
+- Title: ASC / DESC Criterion
+- Start Date: 2021-04-14
+- Specification PR: [#30](https://github.com/meilisearch/specifications/pull/30)
+- MeiliSearch Tracking-Issues: [#161](https://github.com/meilisearch/milli/issues/161)
+
+## 1. Feature Description and Interaction
+
+### I. Summary
+
+Ranking rules are built-in rules that ensure relevancy in search results. Ranking rules are applied in a default order which can be changed in the settings. You can add or remove rules and change their order of importance.
+
+MeiliSearch allows you to create custom rules within the default rules. Custom rules are dedicated to sorting in ascending or descending order on an attribute.
+
+### II. Motivation
+
+We want to provide our users with an always improved usage experience. Relevance is essential in a search engine since it is what allows the engine to fulfill its objective. Delivering results that match user demands by allowing modification of the relevance is critical.
+
+The new search engine called Milli no longer processes this criterion of relevance the same way as the current MeilliSearch. This specification aims to make Milli identical both in API usage and in expected search results as the current ISO version (v0.20).
+
+### III. Additional Materials
+
+#### Algolia
+
+Algolia offers 8 classification rules to achieve relevance.
+
+- Number of typos
+- Geolocation
+- Number of words in the query matching in the result
+- Filters
+- Distance between words
+- Best matching attribute in the record
+- Number of words matching exactly (without typo)
+- Custom ranking
+
+Note that Algolia doesn't recommend changing the order of the default criteria because of the fact that it works for the vast majority of their use cases. Like MeiliSearch (v0.20), documents, where the attribute specified in the custom ranking is missing, are pushed to the bottom of the search results.
+
+### IV.Explanation
+
+#### Current behavior of v0.20
+
+In the case of certain documents that do not contain the attribute configured on a custom ASC/DESC ranking rules, it will return those documents at the bottom of the search result. Documents are not excluded and can be returned in the search result.
```

Comment:
It's not really true. As we discussed, the end of the bucket is returned, meaning you can have this behavior (for a custom ranking rule on `price`):
- 10 results with `price` attributes
- 2 results without `price`
- 8 results with `price`
The end of the bucket is returned.